Sage/carbon
# feat(menu): describe menu using design tokens, update tests - FE-4724

### Proposed behaviour

Describes menu component using the tokens of the design.
Describes focus state using tokens : `box-shadow: inset 0 0 0 var(--borderWidth300) var(--colorsSemanticFocus500)`.
Changes colors of the title segment: 
 - white: `var(--colorsComponentsMenuYin065)`
 - black: `var(--colorsComponentsMenuYang080)`
 - light: `var(--colorsComponentsMenuYin065)`
 - dark: `var(--colorsComponentsMenuYang080)`
 
 Changes the color for icon in search input: 
 - default: `var(--colorsUtilityMajor200)`
 - hover: `var(--colorsUtilityMajor150)`

### Current behaviour

Menu component use the theme styled-component property.
Current focus state is describe by box-shadow: `rgb(255 181 0) 0px 0px 0px 2px inset`.

Current colors of the title segment: 
 - white: `rgb(64, 102, 119)`
 - black: `rgb(153, 173, 182)`
 - light: `rgb(64, 102, 119)`
 - dark: `rgb(153, 173, 182)`
 
Current color for icon in search input: 
 - default: `rgb(140, 163, 173)`
 - hover: `rgb(191, 204, 210)`

### Checklist




#### QA


### Additional context
![image](https://user-images.githubusercontent.com/44143630/151572620-45ee5754-24dc-459c-b71b-fe43209911fd.png)
![image](https://user-images.githubusercontent.com/44143630/151572672-9c114baf-8825-4bfa-a218-29b631e664f7.png)
![image](https://user-images.githubusercontent.com/44143630/151572734-d6f3b5ca-0de5-44b5-984e-93d7b2de4606.png)
![image](https://user-images.githubusercontent.com/44143630/151572806-a3cb5897-45db-4360-9259-7e0e2a8d71b1.png)


### Testing instructions

The correct appearance can be checked in the component: Menu.
Use devtools to check if menu component use css variables.
